### PR TITLE
Update `DataFrame.query` docstring

### LINKED ISSUE #8156: Improve `DataFrame.query` docstring with local variable example
**What happened**:
Prepare Dask distributed:
```
from dask.distributed import Client

client = Client(n_workers=16,
                threads_per_worker=1)
```

This works:
```python
import pandas as pd

data = {'col_1': [3, 2, 1, 0], 'col_2': ['a', 'b', 'c', 'd']}
df = pd.DataFrame.from_dict(data)
pick = 'c'
df.query('col_2 == @pick')
```
This doesn't work:
```python
dd_temp = dd.from_pandas(df, npartitions=2)
dd_temp.query('col_2 == @pick')
```
Throws UndefinedVariableError("local variable 'pick' is not defined").  

**What you expected to happen**:
Dask version of DataFrame.query should work as in pandas.

**Minimal Complete Verifiable Example**:
See above

**Anything else we need to know?**:
No

**Environment**:

- Dask version: 2021.8.0
- Python version: 3.9.6
- Operating System: Linux (Manjaro)
- Install method (conda, pip, source): conda-forge

### PR DESCRIPTION
**Pull request by pavithraes**
- [x] Closes #8156
- [x] ~Tests added / passed~
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -4827,6 +4838,46 @@ def query(self, expr, **kwargs):
...
+        1  2  2    3
+        2  1  3    2
+
+        Refer to variable name using f-strigs:
+
+        >>> value = 1
+        >>> ddf.query(f'x == {value}').compute()
+            x  y  z z
+        0  1  1    4
+        2  1  3    2
```
Comment (jrbourbeau) [inline dask/dataframe/core.py:4873]: The `x y z z` header needs to be moved over to the left a bit so it aligns with the data below (similar to the two `query` examples above).

**Review by jrbourbeau — COMMENTED** [review]
Thanks @pavithraes! Overall this looks great. There's a small change that needs to be made to make the doctest CI build happy

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -4827,6 +4838,46 @@ def query(self, expr, **kwargs):
...
+        >>> ddf = dd.from_pandas(df, npartitions=2)
+
+        Refer to column names directly:
+
+        >>> ddf.query('y > x').compute()
+           x  y  z z
+        2  1  3    2
+        3  2  4    1
+
+        Refer to column name using `backticks`:
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply dask/dataframe/core.py]: Thanks for the catch! Not intentional, just removed it!

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -4827,6 +4838,46 @@ def query(self, expr, **kwargs):
...
+        >>> value = 1
+        >>> ddf.query(f'x == {value}').compute()
+            x  y  z z
+        0  1  1    4
+        2  1  3    2
+
+        Refer to variable name using ``local_dict``:
+
+        >>> ddf.query('x == @value', local_dict={"value": value}).compute()
+            x  y  z z
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply dask/dataframe/core.py]: Indeed, thanks!

**Review by jrbourbeau — APPROVED** [review]
Thanks for the updates -- will merge after CI finishes 👍

**Pull request merged**